frontend/catalyst/utils/gen_mlir.py

```diff
@@ -21,18 +21,20 @@
 from jaxlib.mlir.dialects._func_ops_gen import FuncOp
 
 
-def gen_setup(ctx):
+def gen_setup(runtime, ctx):
     """
     This function returns an MLIR module with the "setup" function. The setup
     function is a function that needs to be called before calling a
     JIT-compiled function. It initializes the global device context in the runtime.
     """
     txt = """
-func.func @setup() -> () {
+func.func @setup() -> () {{
+    "quantum.device"() {{ specs = ["backend", "{runtime}"] }} : () -> ()
```

Comment:
Having said that, this really only makes sense where the runtime and execution device have the same scope.

In a scenario where we are aiming for devices to be scoped to a specific QNode, it would make more sense to have such a device instruction inserted to the start of each QNode body.

Reply:
Although we may get rid of these `init` and `finalize` ops down the road, the scope of a QNode currently is between these two instructions and I supposed one should be able to specify one-to-many device specifications before the initialization for every QNode and `__rt__device` is simply trying to mimic the concept of `qml.device` and `qml.qnode` in the IR.

Reply:
To summarize after an offline discussion, the `init` and `finalize` instructions don't indicate the scope of a QNode, but rather the scope of a QJIT unit which contains general code than call into one or multiple QNodes, so we'll move the location of the device setup instruction to be scoped to each QNode.